platform-sdk/swirlds-common/src/main/java/com/swirlds/common/stream/EventStreamManager.java

```diff
@@ -197,119 +202,60 @@ public void stop() {
-    }
-
     /**
-     * receives a consensus event from ConsensusRoundHandler each time, sends it to multiStream which then sends to two
-     * queueThread for calculating runningHash and writing to file
+     * Adds a list of events to the event stream.
      *
-     * @param event the consensus event to be added
+     * @param events the list of events to add
      */
-    public void addEvent(final T event) {
-        if (!freezePeriodStarted) {
-            multiStream.addObject(event);
-            if (isLastEventInFreezeCheck.test(event)) {
-                freezePeriodStarted = true;
-                logger.info(
-                        EVENT_STREAM.getMarker(),
-                        "ConsensusTimestamp of the last Event to be written into file before restarting: " + "{}",
-                        event::getTimestamp);
-                multiStream.close();
+    public void addEvents(@NonNull final List<T> events) {
+        events.forEach(event -> {
+            if (!freezePeriodStarted) {
+                multiStream.addObject(event);
+                if (isLastEventInFreezeCheck.test(event)) {
+                    freezePeriodStarted = true;
+                    logger.info(
+                            EVENT_STREAM.getMarker(),
+                            "ConsensusTimestamp of the last Event to be written into file before restarting: {}",
+                            event::getTimestamp);
+                    multiStream.close();
+                }
+            } else {
+                eventAfterFreezeLogger.warn(
+                        EVENT_STREAM.getMarker(), "Event {} dropped after freezePeriodStarted!", event.getTimestamp());
             }
-        } else {
-            eventAfterFreezeLogger.warn(
-                    EVENT_STREAM.getMarker(), "Event {} dropped after freezePeriodStarted!", event.getTimestamp());
-        }
+        });
     }
 
     /**
-     * sets startWriteAtCompleteWindow: it should be set to be true after reconnect, or at state recovering; it should
-     * be set to be false at restart
+     * Updates the running hash with the given event hash. Called when a state is loaded.
      *
-     * @param startWriteAtCompleteWindow whether the writer should not write until the first complete window
+     * @param runningEventHashUpdate the hash to update the running hash with
      */
-    public void setStartWriteAtCompleteWindow(final boolean startWriteAtCompleteWindow) {
+    public void updateRunningHash(@NonNull final RunningEventHashUpdate runningEventHashUpdate) {
         if (streamFileWriter != null) {
-            streamFileWriter.setStartWriteAtCompleteWindow(startWriteAtCompleteWindow);
+            streamFileWriter.setStartWriteAtCompleteWindow(runningEventHashUpdate.isReconnect());
         }
+
+        initialHash = new Hash(runningEventHashUpdate.runningEventHash());
+        logger.info(EVENT_STREAM.getMarker(), "EventStreamManager::updateRunningHash: {}", () -> initialHash);
```

Comment:
Are you sure the `{}` will capture the return value of the lambda and not the lambda itself? If not, then I believe that `toString()` on a lambda itself is hardly useful for diagnostics. But if yes, then why not pass the `initialHash` value directly as a parameter here?

Reply:
I actually see no reason for a lambda if we are just passing an object

Reply:
Yeah, I think the object should just be passed directly, no reason for a lambda

Reply:
This was old code just reorganized - removing the lambda